test/specs/requests.spec.js

```diff
@@ -157,6 +157,26 @@ describe('requests', function () {
     });
   });
 
+  it('When the file protocol is initiated, it should reolve when response.status is equal to 0', function (done) {
+    var resolveSpy = jasmine.createSpy('resolve');
+    var rejectSpy = jasmine.createSpy('reject');
+
+    axios('./foo').then(resolveSpy)
+      .catch(rejectSpy)
+      .then(function () {
+        expect(resolveSpy).toHaveBeenCalled();
+        expect(rejectSpy).not.toHaveBeenCalled();
+        done();
+      });
+
+    getAjaxRequest().then(function (request) {
+      request.respondWith({
+        status: 0,
+        responseURL: 'file:///c://project//test.txt',
```

Comment:
It should be axios('file://xx'), instead of here.

Reply:
Thanks for your review, updated code.

Reply:
@Alanscut I mean, line 164 should be changed and line 175 can be removed. 😆

Reply:
@chinesedfanLine 175 cannot be removed, otherwise 'request.responseURL' is 'null', which will result in failure to execute the 'settle' function and cause a timeout.

### xhr.js

https://github.com/axios/axios/blob/master/lib/adapters/xhr.js#L45
```javascript
 if (request.status === 0 && !(request.responseURL && request.responseURL.indexOf('file:') === 0)) {
       return;
 }

```

Reply:
@Alanscut Not really. It is short-circuit evaluation, and I have tested locally. But you can also leave it there.